### PR TITLE
Increase the stability of the computation of Det, and add LogDet func…

### PR DESCRIPTION
**Pull request by btracey**
…tion to enable enhanced numeric stability

### CODE REVIEW

**Comment by btracey** [comment]
Using LogSumExp is much better than multiplying the elements directly for computing the determinant (for example if there is one very large diagonal element and another very small element).

 I could use some help wordsmithing the explanation for why LogDet. Many matrix expressions have something like const / abs(det(A)). It's much better to do exp( log(const) - abs(logdet)) than it is to try and compute the determinant explicitly. This is the sort of case I am trying to describe.

File: mat64/lu.go [hunk 1]
```diff
@@ -68,14 +69,29 @@ func (lu *LU) Factorize(a Matrix) {
...
-// Det returns the determinant of the matrix that has been factorized.
+// Det returns the determinant of the matrix that has been factorized. In many
+// expressions, using LogDet will be more numerically stable.
 func (lu *LU) Det() float64 {
+	det, sign := lu.LogDet()
+	return math.Exp(det) * sign
+}
+
+// LogDet returns the log of the determinant and the sign of the determinant
+// for the matrix that has been factorized. Many expressions multiply or divide
```
Comment (kortschak) [inline mat64/lu.go]: I'm not sure that this needs to be spelled out, but:

"Numerical stability in product and division expressions is generally improved by working in log space."

**Comment by kortschak** [comment]
LGTM

**Comment by kortschak** [comment]
Updates #138.

**Pull request merged**